### PR TITLE
Fix fsgroup

### PR DESCRIPTION
**Pull request by lifupan**
fix the issue of missing pass/set fsGroup for emptyDir volumes
    
For k8s emptyDir volume, a specific fsGroup would
be set for it, thus runtime should pass this fsGroup
to guest and set it properly on the emptyDir volume
in guest.
    
Fixes: #1580

### CODE REVIEW

File: src/runtime/virtcontainers/kata_agent.go [hunk 1]
```diff
@@ -1426,10 +1433,27 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) []*grpc.Storage
...
-func (k *kataAgent) handleLocalStorage(mounts []specs.Mount, sandboxID string, rootfsSuffix string) []*grpc.Storage {
+func (k *kataAgent) handleLocalStorage(mounts []specs.Mount, sandboxID string, rootfsSuffix string) ([]*grpc.Storage, error) {
 	var localStorages []*grpc.Storage
 	for idx, mnt := range mounts {
 		if mnt.Type == KataLocalDevType {
+			origin_src := mounts[idx].Source
+			stat := syscall.Stat_t{}
+			err := syscall.Stat(origin_src, &stat)
+			if err != nil {
+				k.Logger().Errorf("failed to stat %s", origin_src)
```
Comment (liubin) [inline src/runtime/virtcontainers/kata_agent.go:1444]: Nit: Add `WithError(err)`

Comment (lifupan) [reply src/runtime/virtcontainers/kata_agent.go]: Fixed.

**Review by liubin — APPROVED** [review]

**Comment by wainersm** [comment]
Hi @lifupan ,

As I commented on #1580, for one out of three test cases I could verify this fix. Because the CI job on openshift-ci is broken I couldn't check the two test cases, nevertheless feel free to merge this PR. I will manage to run the other tests, in case they don't pass I can contact you so we look for further fixes.

**Review by Tim-Zhang — APPROVED** [review]

**Pull request merged**